hooks/useChannelStake/useStake.ts

```diff
@@ -0,0 +1,41 @@
+import { useToast } from "@/components/ui/use-toast";
+import { executeStake } from "@/lib/sonata/staking";
+import { useNeynarProvider } from "@/providers/NeynarProvider";
+import { useStakeProvider } from "@/providers/StakeProvider";
+import { useTipProvider } from "@/providers/TipProvider";
+import { useParams } from "next/navigation";
+
+const useStake = () => {
+  const { channelId } = useParams();
+  const { toast } = useToast();
+  const { signer } = useNeynarProvider();
+  const { balance, setBalance } = useTipProvider();
+  const { setChannelDetails, channelDetails, setUserStakedAmount, userStakedAmount } = useStakeProvider();
+
+  return async (amount: number) => {
```

Comment:
could you update this hook to
1. define this as the `stake` function 
2. return `{stake}` to match our return values from other hooks